Compiler crash when returning enum instead of boolean

<!--
 Thanks for submitting an issue to AssemblyScript! Please take a moment to
 read the contributing guidelines linked below to get off to a good start 🙂
-->

### The issue
This is erroneous code, but it crashes the compiler:
```ts
enum Bar {
  Baz
}

export function foo(): boolean {
  return Bar.Baz;
}

foo();
```

### The stack trace
```
▌ Whoops, the AssemblyScript compiler has crashed during compile :-(
▌ 
▌ Here is a stack trace that may or may not be useful:
▌ 
▌ AssertionError: assertion failed
▌     at assert (D:\assemblyscript\std\portable\index.js:199:9)
▌     at Flow.canOverflow (D:\assemblyscript\src\flow.ts:1221:9)
▌     at Compiler.ensureSmallIntegerWrap (D:\assemblyscript\src\compiler.ts:10023:18)
▌     at Compiler.compileExpression (D:\assemblyscript\src\compiler.ts:3544:27)
▌     at Compiler.compileReturnStatement (D:\assemblyscript\src\compiler.ts:2820:19)
▌     at Compiler.compileStatement (D:\assemblyscript\src\compiler.ts:2209:21)
▌     at Compiler.compileStatements (D:\assemblyscript\src\compiler.ts:2273:23)
▌     at Compiler.compileFunctionBody (D:\assemblyscript\src\compiler.ts:1609:20)
▌     at Compiler.compileFunction (D:\assemblyscript\src\compiler.ts:1539:17)
▌     at Compiler.makeCallDirect (D:\assemblyscript\src\compiler.ts:7211:15)
▌     at Compiler.compileCallDirect (D:\assemblyscript\src\compiler.ts:6699:17)
▌     at Compiler.compileCallExpression (D:\assemblyscript\src\compiler.ts:6314:21)
▌     at Compiler.compileExpression (D:\assemblyscript\src\compiler.ts:3459:21)
▌     at Compiler.compileExpressionStatement (D:\assemblyscript\src\compiler.ts:2496:17)
▌     at Compiler.compileStatement (D:\assemblyscript\src\compiler.ts:2193:21)
▌ 
▌ If it refers to the dist files, try to 'npm install source-map-support' and
▌ run again, which should then show the actual code location in the sources.
▌ 
▌ If you see where the error is, feel free to send us a pull request. If not,
▌ please let us know: https://github.com/AssemblyScript/assemblyscript/issues
▌ 
▌ Thank you!
```

### Credits
Thanks to @fabriciopashaj for crashing the compiler

--- a/src/flow.ts
+++ b/src/flow.ts
@@ -26,9 +26,9 @@ import {
   Function,
   Element,
   ElementKind,
-  Global,
   Field,
-  Class
+  Class,
+  TypedElement
 } from "./program";
 
 import {
@@ -1218,8 +1218,8 @@ export class Flow {
       case ExpressionId.GlobalGet: {
         // TODO: this is inefficient because it has to read a string
         let global = assert(this.parentFunction.program.elementsByName.get(assert(getGlobalGetName(expr))));
-        assert(global.kind == ElementKind.GLOBAL);
-        return canConversionOverflow((<Global>global).type, type);
+        assert(global.kind == ElementKind.GLOBAL || global.kind == ElementKind.ENUMVALUE);
+        return canConversionOverflow((<TypedElement>global).type, type);
       }
 
       case ExpressionId.Binary: {

--- a/tests/compiler/enum-return-error.json
+++ b/tests/compiler/enum-return-error.json
@@ -1,0 +1,8 @@
+{
+  "asc_flags": [
+   ],
+  "stderr": [
+    "AS200: Conversion from type 'i32' to 'bool' requires an explicit cast.",
+    "EOF"
+  ]
+}

--- a/tests/compiler/enum-return-error.ts
+++ b/tests/compiler/enum-return-error.ts
@@ -1,0 +1,11 @@
+enum Bar {
+  Baz
+}
+
+function foo(): boolean {
+  return Bar.Baz;
+}
+
+foo();
+
+ERROR("EOF");

